### PR TITLE
Fix dispatch when there are different receivers

### PR DESCRIPTION
**Pull request by jkeiser**
with different numbers of arguments.
Fixes https://github.com/chef/chef-dk/issues/546

### CODE REVIEW

**Comment by danielsdeleo** [comment]
Wow, nice catch. :+1:

**Comment by danielsdeleo** [comment]
@chef/client-core This is a blocker for 12.5 final

**Comment by coderanger** [comment]
I'm a little confused what this actually changed?

File: lib/chef/event_dispatch/dispatcher.rb [hunk 1]
```diff
@@ -32,8 +32,11 @@ def call_subscribers(method_name, *args)
           mth = s.method(method_name)
           # Trim arguments to match what the subscriber expects to allow
           # adding new arguments without breaking compat.
-          args = args.take(mth.arity) if mth.arity < args.size && mth.arity >= 0
-          mth.call(*args)
+          if mth.arity < args.size && mth.arity >= 0
```
Comment (chefsalim) [inline lib/chef/event_dispatch/dispatcher.rb]: hmm, seems like the same if expression?

**Comment by jkeiser** [comment]
@coderanger the issue was that we were modifying the input `args` variable before; so if we truncated them to pass to a formatter that took 1 arg, then we'd have to pass 1 args to everyone (even those who expect 2).

File: lib/chef/event_dispatch/dispatcher.rb (hunk 1 above)
Comment (jkeiser) [inline lib/chef/event_dispatch/dispatcher.rb]: It is indeed! The only change is we don't modify `args`.

**Comment by jkeiser** [comment]
The attached tests fail without the change, and succeed with them, fyi.

**Comment by jaym** [comment]
:+1:

**Comment by coderanger** [comment]
Got it :+1:  :)

**Comment by lamont-granquist** [comment]
:+1: FWIW

**Pull request merged**